bin/list-nginx-class-ingresses.rb

```diff
@@ -3,42 +3,42 @@
 # Script to build the list of ingresses and its namespaces with
 # the ingress-class annotations as "nginx" or empty i.e "" or ingress
 # with no annotations and into a json file
+# This script will list ingress for all non-prod namespaces. For production ingresses
+# change if non_production?(namespace) to if production?(namespace)
 
 require "json"
 require "open3"
 
 NGINX_CLASS_INGRESS_LIST_FILE = "nginx_class_ingresses.json"
 
 def main
-  nginx_class_ingresses = list_ingresses_has_annotations("nginx")
-  null_class_ingresses = list_ingresses_has_annotations("")
-  no_annotation_ingresses = list_ingresses_no_annotations
-  list = nginx_class_ingresses + null_class_ingresses + no_annotation_ingresses
+  $namespaces = get_namespaces
+  nginx_class_ingresses = ingresses_matching_class("nginx")
+  null_class_ingresses = ingresses_matching_class("")
+  no_annotation_ingresses = ingresses_matching_no_class
 
-  File.write(NGINX_CLASS_INGRESS_LIST_FILE, list.to_json)
+  list = nginx_class_ingresses.compact + null_class_ingresses.compact + no_annotation_ingresses.compact
```

Comment:
You can do this as:
```
list = (nginx_class_ingresses + null_class_ingresses + no_annotation_ingresses).compact
```
...which is slightly cleaner

Reply:
Updated the code